components/reference-implementation-openbanking-nextgenpsd2/src/main/java/org/wso2/openbanking/nextgenpsd2/extensions/api/impl/ConsentManagementAPIImpl.java

```diff
@@ -0,0 +1,228 @@
+                return Response.status(Response.Status.BAD_REQUEST).entity(errorResponse).build();
+            }
+
+            return Response.ok().entity(new JSONObject(validationResponse).toString()).build();
+
+        } catch (BadRequestException | ServerErrorException e) {
+            log.error("[" + requestId + "] " + "An error occurred creating consent.", e);
+            return Response.status(e.getStatus()).entity(e.getFormattedErrorAsString()).build();
+        } catch (org.wso2.openbanking.nextgenpsd2.extensions.exceptions.ApiException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Method for returning the response for pre-processing consent creation request.
+     *
+     * @param requestBody
+     * @return
+     */
+    public static Response preProcessConsentCreation(PreProcessConsentCreationRequestBody requestBody) {
+        String requestId = requestBody.getRequestId();
+        try {
+            // Validate X-request-ID header
+            // Enable forwarding of the specific header in accelerator configurations
+            CommonConsentValidationUtil.validateRequestIdentificationHeader(requestBody.getData().getRequestHeaders());
+
+            // Extract consent initiation data
+            JSONObject consentInitiationDataJSON;
+            try {
+                consentInitiationDataJSON = CommonConsentValidationUtil.convertObjectToJson(requestBody.getData()
+                        .getConsentInitiationData());
+
+                if (consentInitiationDataJSON.isEmpty()) {
+                    // If payload is empty
+                    JSONObject errorResponse = ErrorUtil.getFormattedFailedResponse(400,
+                            ErrorUtil.constructBerlinError(null, TPPMessage.CategoryEnum.ERROR,
+                                    TPPMessage.CodeEnum.FORMAT_ERROR, ErrorConstants.PAYLOAD_NOT_PRESENT_ERROR));
+                    return Response.ok().entity(errorResponse.toString()).build();
+                }
+
+            } catch (JSONException e) {
+                // If payload is not JSON
+                JSONObject errorResponse = ErrorUtil.getFormattedFailedResponse(400,
+                        ErrorUtil.constructBerlinError(null, TPPMessage.CategoryEnum.ERROR,
+                                TPPMessage.CodeEnum.FORMAT_ERROR, ErrorConstants.PAYLOAD_FORMAT_ERROR));
+                return Response.status(Response.Status.OK).entity(errorResponse.toString()).build();
+            }
+
+            ConsentManagementValidationHandler consentManagementValidationHandler = CommonConsentValidationUtil
+                    .getConsentManagementResponseHandler(requestBody.getData().getConsentResourcePath());
+
+            SuccessResponsePreProcessConsentCreation validationResponse;
+
+            if (consentManagementValidationHandler != null) {
+                validationResponse = consentManagementValidationHandler.handleCreation(requestBody);
```

Comment:
Shall we rename the `validationResponse` variable name to `consentCreationResponse` or something?